Thermal optics tweak: less night vision, more flash protection.

AcapitalA opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request
This changes thermal optics to have no negative flash protection while making it slightly worse for use as night vision goggles. Balance-wise it stays about the same, but puts thermals in a more seperate place rather than sharing similarities with night vision.

NVGs:
![SOUL](https://user-images.githubusercontent.com/52262416/63173962-471cfc80-c041-11e9-9bcf-ea16980b6a30.png)
Thermals:
![SOULLESS](https://user-images.githubusercontent.com/52262416/63174021-6582f800-c041-11e9-8fa6-c02ba351268d.png)
<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game
This change puts thermal optics apart from night vision optics by making thermals more unique on their own. This would increase the variety we have on special optics by a bit.

<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->

## Changelog
:cl:
balance: Thermal optics got tweaked to have no negative flash protection in exchange for slightly less vision in the dark.
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

AcapitalA committed:
Creates MILDLY_VISIBLE (224) for thermal optics

Thermal optics get their own define for balance.

AcapitalA committed:
Changes thermal eyes.

Thermal eyes now have MILDLY_VISIBLE and their negative flash protection was removed.

AcapitalA committed:
Adds MILDLY_VISIBLE to thermal glasses.

Also updated darkness_view to 8 so it doesn't look weird.

AcapitalA committed:
Thermal Toggle gets MILDLY_VISIBLE.

Just in case.

Fikou commented:
what about genetics thermal mutation

AcapitalA commented:
> what about genetics thermal mutation

Crap, completely forgot about the genetics mutation, it's so obscure to find for me but I'll fix that one ASAP.

necromanceranne commented:
I don't suppose you could give us a visual representation of what it would look like could you?

AcapitalA committed:
Thermal vision from genetics is affected too.

MILDLY_VISIBLE added to the genetics mutation. - Genetics code is all over the place, jesus christ.

AcapitalA commented:
> I don't suppose you could give us a visual representation of what it would look like could you?

I can try, I'll start up localhost and see if i can get some good pictures for it. I'll throw them in the  OP.

ExcessiveUseOfCobblestone commented:
`verified`

Ratvar commented:
"Balance-wise it stays about the same"
You still see everyone and everything, how's that the same? Buff and nerf don't make even.

AcapitalA commented:
> "Balance-wise it stays about the same"
> You still see everyone and everything, how's that the same? Buff and nerf don't make even.

You still see everyone through walls, and you can still see in the dark but it's a lot less compared to night vision for, well, night vision. Night vision goggles are mass printable and allow you to see more clearly in the dark being the exchange here.

Akrilla commented:
Thermal vision is the xray right? Okay, so it's now not also night vision - I don't see why this is getting 👎 'd

AcapitalA commented:
> Thermal vision is the xray right? Okay, so it's now not also night vision - I don't see why this is getting 👎 'd

X-ray vision is the one where you can see everything through walls.
Thermal vision is the one where you can see people through walls.

I'm planning to do something about X-ray somewhere later on, so people don't go for it all the time and rather get to pick between them. This PR just seperates thermals more from night vision.

Akrilla commented:
Oh then even bigger +1 then

optimumtact commented:
Power creep

AcapitalA commented:
> Power creep

At some point I was thinking about removing the night vision entirely and make people through walls 100% visible instead, but I kept it this way considering people were more for the latter idea.

In the end, I'm just trying to get people to pick something else than X-ray because currently, you'll very rarely hear anyone not pick X-ray over any of the others.

I'm also looking into fixing X-ray somewhat, considering literal wallhacks are very strong in general.

SkeletalElite commented:
> At some point I was thinking about removing the night vision entirely and make people through walls 100% visible instead, but I kept it this way considering people were more for the latter idea.

I'd rather it kept the flash weakness than lose the ability to see in the dark, seeing in the dark is half the reason to use thermals. I don't mind it being harder to see in the dark than night vision, just as long as it's still possible.

AcapitalA commented:
> At some point I was thinking about removing the night vision entirely and make people through walls 100% visible instead, but I kept it this way considering people were more for the latter idea.
> 
> I'd rather it kept the flash weakness than lose the ability to see in the dark, seeing in the dark is half the reason to use thermals. I don't mind it being harder to see in the dark than night vision, just as long as it's still possible.

I've heard lots of people like the "see in the dark" aspect of thermals, and that's why I decided to reduce it instead of completely removing it. I think I can still come up with some ideas to keep the optics mostly seperate from eachother regardless, so it shouldn't be much of an issue.

wesoda25 commented:
When the flap is thermal vision gonna be coded so that it displays people the way actual thermal vision works. It’d be especially cool if normal body temp is displayed as orange, on fire is red and frozen is yellow-blue

wesoda25 commented:
Also completely remove the see in the dark functionality. It makes no sense and is bad gameplay, especially if theres no flash weakness anymore.

carlarctg commented:
removing the minor NV thermals give will make it a lot harder to go around in maintenance/dark areas without showing your position when hunting with thermals. the way light works people will instantly know you're there which ruins half the point of using thermals in a sneaky manner.

knacker48 commented:
The small amount of NV that thermal vision gave it a reason to be taken rather be just a worse version of another power. If you get rid of that then no-one is gonna want to take it with x-ray (or more likely at all), thereby removing a possible option for those who want to experiment with different powers

wesoda25 commented:
No the reason to take thermals is to see people through walls, not to remove the need for a flashlight in maint.

ATH1909 commented:
Personally, I _like_ how the thermal vision genetics power gives you night vision; it makes for a nice combo with x-ray.

If you're gonna nerf thermal vision this badly (I can see approximately jack !@#$ in that second image), could you at least give genetics a night vision power to compensate for its loss?

PKPenguin321 commented:
I don't think this touches the genetics powers

bigfatbananacyclops commented:
it would be cool if there was literally a thermal vision added to the goggles, as in, you can see the heat mobs and everything else is cold, i always found it dumb that you get more eye damage from flashing with thermals.

actioninja commented:
Not really a fan of this. The "nerf" doesn't really nerf at all and I can still see pretty clearly in the reduced brightness, and on top of that it's easily abusable with the right click menu.
